src/MedTrackerAPI/Features/Devices/GetDevice.cs

```diff
@@ -0,0 +1,45 @@
+using MediatR;
+using MedTrackerAPI.Endpoints;
+using MedTrackerAPI.Infrastructure;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace MedTrackerAPI.Features.Devices;
+
+public static class GetDevice
+{
+    public record Response(int DeviceId, string Description, string Manufacturer, string Model, string PartNumber, string LotNumber);
+    
+    public sealed class Endpoint : IEndpoint
+    {
+        public void MapEndpoint(IEndpointRouteBuilder app)
+        {
+            app.MapGet("devices/{deviceId}", async (IMediator mediator, int deviceId) =>
+                {
+                    var query = new GetDeviceQuery { DeviceId = deviceId };
+                    var device = await mediator.Send(query);
+                    
+                    return device is null ? Results.BadRequest("Device not found") : Results.Ok(device);
```

Comment:
Um ackshually, if you're wanting to follow the HTTP spec this would be a 404 (Not Found), not a 400 (Bad Request). 404's don't usually have a body to them, with the 404 being enough to convey "Not found"